controllers/articles-controllers.js

```diff
@@ -16,9 +17,21 @@ function getArticleById(request, response, next) {
 }
 
 function getArticles(request, response, next) {
-  fetchAllArticles().then((articles) => {
-    response.status(200).send(articles);
-  });
+  const { topic } = request.query;
+
+  if (topic) {
+    return fetchArticleByTopic(topic)
+      .then((articles) => {
+        response.status(200).send(articles);
```

Comment:
remember that we ideally want to be sending our data back as an object (JSON) - would be better to send back our articles data in an object with a descriptive key, similarly to how you're currently sending your error message back on a key of message, not just as a string